homeassistant/components/smartthings/sensor.py

```diff
@@ -953,14 +959,20 @@ async def async_setup_entry(
     async_add_entities(
         SmartThingsSensor(entry_data.client, device, description, capability, attribute)
         for device in entry_data.devices.values()
-        for capability, attributes in device.status[MAIN].items()
-        if capability in CAPABILITY_TO_SENSORS
-        for attribute in attributes
-        for description in CAPABILITY_TO_SENSORS[capability].get(attribute, [])
-        if not description.capability_ignore_list
-        or not any(
-            all(capability in device.status[MAIN] for capability in capability_list)
-            for capability_list in description.capability_ignore_list
+        for capability, attributes in CAPABILITY_TO_SENSORS.items()
+        if capability in device.status[MAIN]
+        for attribute, descriptions in attributes.items()
+        for description in descriptions
+        if (
+            not description.capability_ignore_list
+            or not any(
+                all(capability in device.status[MAIN] for capability in capability_list)
+                for capability_list in description.capability_ignore_list
+            )
+        )
+        and (
+            not description.except_if_state_none
+            or device.status[MAIN][capability][attribute].value is not None
```

Comment:
So since the library now provides `Capability | str` this code wasn't happy as we did dict lookups with `str` while the key of that dict is a `Capability`. We will never support an entity in the integration that is not a `Capability`.

So I now rewrote this to iterate over the list we have in the integration instead of what we get from the device.

This uncovered a bug where `powerConsumptionReport` can be `None` and then the `value_fn` fails. So I updated the entity description to add a. check if the value of th attribute we try to use is `None`

Reply:
Is it safe that `powerConsumptionReport` will not be `None` after you already added the entity?

Reply:
That won't happen. It's either `None` or a `dict`.